x-pack/plugins/security_solution/cypress/objects/rule.ts

```diff
@@ -41,7 +41,8 @@ export interface CustomRule {
   customQuery?: string;
   name: string;
   description: string;
-  index: string[];
+  index?: string[];
+  dataView?: string;
```

Comment:
I think there's a problem with this model. I think it should reflect the following invariants:

- ML rules don't have `index` or `dataView`. I think this is expressed correctly in the code because `MachineLearningRule` doesn't extend `CustomRule`
- Other rule types have _either_ `index` _or_ `dataView`. This is expressed incorrectly because this model allows a rule to have _neither_ of them.

Could you please try this model and see if it works?

```ts
type RuleDataSource =
  | { type: 'indexPatterns'; index?: string[] }
  | { type: 'dataView'; dataView?: string };

export interface CustomRule {
  customQuery?: string;
  name: string;
  description: string;
  dataSource: RuleDataSource;
  interval?: string;
  severity: string;
  riskScore: string;
  tags: string[];
  timelineTemplate?: string;
  referenceUrls: string[];
  falsePositivesExamples: string[];
  mitre: Mitre[];
  note: string;
  runsEvery: Interval;
  lookBack: Interval;
  timeline: CompleteTimeline;
  maxSignals: number;
  buildingBlockType?: string;
  exceptionLists?: Array<{ id: string; list_id: string; type: string; namespace_type: string }>;
}
```